client/state/data-layer/wpcom/sites/activity/index.js

```diff
@@ -1,29 +1,45 @@
+/** @format */
 /**
  * External dependencies
  */
-import { pick } from 'lodash';
+import { get, pick } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import { http } from 'state/data-layer/wpcom-http/actions';
 import { ACTIVITY_LOG_REQUEST } from 'state/action-types';
-import {
-	activityLogError,
-	activityLogUpdate,
-} from 'state/activity-log/actions';
+import { activityLogError, activityLogUpdate } from 'state/activity-log/actions';
+
+const KNOWN_PARAMS = [ 'action', 'date_end', 'date_start', 'group', 'name', 'number' ];
 
 export const handleActivityLogRequest = ( { dispatch }, action ) => {
-	dispatch( http( {
-		apiVersion: '1',
-		method: 'GET',
-		path: `/sites/${ action.siteId }/activity`,
-		query: {
-			number: 1000,
-			...action.params,
-		},
-	}, action ) );
+	const { siteId } = action;
+
+	const query = pick( action.params, KNOWN_PARAMS );
+
+	const dateEnd = get( action.params, 'dateEnd' );
+	if ( dateEnd ) {
+		query.date_end = dateEnd;
+	}
+
+	const dateStart = get( action.params, 'dateStart' );
```

Comment:
This will get repetitive if we do it for more of the query params - should we loop through all of the action params and add them to the query, using a helper object that maps `dateStart` to `date_start` for example?